src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/FishboneMoncriefDisk.cpp

```diff
@@ -134,6 +138,9 @@ FishboneMoncriefDisk::IntermediateVariables<DataType, NeedSpacetime>::
   z_squared /= -r_squared;
   z_squared += 1.0;
   sin_theta_squared = std::move(z_squared);
+  // Because of the subtraction done to compute sin^2(theta) can be negative by
+  // roundoff. This is a fix for that for now.
```

Comment:
What makes this fix a temporary one? Is there a better way? If so, can you make an issue so that we don't forget?

Reply:
I don't know what the best way to compute these quantities is. Someone should think about it and we should do that :) I'll add an issue!

Reply:
sin(theta)^2 = X / r^2

where X is the nonnegative root of

0 = X^2 - (x^2 + y^2 - z^2 - a^2) X - (x^2 + y^2) z^2

Reply:
(formula verified in code review) :+1:
Note also that the -z^2 in the second term is correct as written (as may be surprising on a first reading).

Reply:
#1270